src/components/ble/DfuService.cpp

```diff
@@ -388,6 +388,7 @@ void DfuService::DfuImage::WriteMagicNumber() {
 }
 
 void DfuService::DfuImage::Erase() {
+  spiNorFlash.Wakeup();
```

Comment:
This should not be needed, as `SystemTask` already takes care to wake up all devices, the SPI NOR flash included.
Did you spot any issue without waking the memory here?

Reply:
Without this, future OTA DFU would fail if started with the screen turned off.

I see that the [`BleFirmwareUpdateStart` message is pushed](https://github.com/JF002/InfiniTime/blob/883700fca13a45b6cb0af4d74e3ef276a2d69283/src/components/ble/DfuService.cpp#L208) before going to the [`Start` state](https://github.com/JF002/InfiniTime/blob/883700fca13a45b6cb0af4d74e3ef276a2d69283/src/components/ble/DfuService.cpp#L116) where the SPI flash is erased. So my guess is that this pushed message, or [the `GotoRunning` message](https://github.com/JF002/InfiniTime/blob/883700fca13a45b6cb0af4d74e3ef276a2d69283/src/systemtask/SystemTask.cpp#L382), is not processed before arriving at the flash erase step.

Reply:
Thanks for the explanation! You are right : the message `GotoRunning` is handled by SystemTask, while the DFU service is handled by the BLE task. 
On my setup, SystemTask wakes the memory up before the BLE task tries to erase the flash memory but... nothing prevent the BLE task to run before SystemTask (which is probably what you experienced).

So yeah, calling Wakeup() in DFUService might seem a good idea but there might still be a risk of race condition (both systemtask and ble task call that method) and wakeup will be called twice...

I wondering if we can find a nice and explicit solution to ensure that DFU does not use the memory until it's woken up by SystemTask.
Or ensure that the priorities of the tasks are correctly set to ensure that this race condition won't happen ?